model/project.go

```diff
@@ -1441,28 +1467,55 @@ func (p *Project) FindAllBuildVariantTasks() []BuildVariantTaskUnit {
 	allBVTs := []BuildVariantTaskUnit{}
 	for _, b := range p.BuildVariants {
 		for _, t := range b.Tasks {
-			tasksToPopulate := []ProjectTask{}
+			t.Variant = b.Name
 			if t.IsGroup {
 				group, exists := taskGroups[t.Name]
 				if !exists {
 					continue
 				}
-				for _, groupTask := range group.Tasks {
-					tasksToPopulate = append(tasksToPopulate, tasksByName[groupTask])
-				}
+				allBVTs = append(allBVTs, p.TasksFromGroup(t, group)...)
 			} else {
-				tasksToPopulate = append(tasksToPopulate, tasksByName[t.Name])
-			}
-			for _, pTask := range tasksToPopulate {
-				t.Populate(pTask)
-				t.Variant = b.Name
+				t.Populate(tasksByName[t.Name])
 				allBVTs = append(allBVTs, t)
 			}
 		}
 	}
 	return allBVTs
 }
 
+// TasksFromGroup returns a slice of the task group's tasks. groupTask is merged with each of the tasks in the group.
+func (p *Project) TasksFromGroup(groupTask BuildVariantTaskUnit, tg TaskGroup) []BuildVariantTaskUnit {
```

Comment:
nit: Can "groupTask" be renamed to taskGroupTask or bvTaskGroup? I think we don't typically abbreviate to group (except in this function name I guess haha) so this did confuse me

Reply:
Could we also  potentially only pass in tg.Tasks as taskGroupTaskNames or something instead of the whole task group?

Reply:
Last comment here sorry 😅  but this function should also be private.

Reply:
I rearranged it to just take the `bvTaskGroup`